include/sparrow/arrow_interface/array_data_to_arrow_array_converters.hpp

```diff
@@ -0,0 +1,123 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <algorithm>
+#include <cstdint>
+#include <type_traits>
+
+#include "sparrow/array/array_data.hpp"
+#include "sparrow/arrow_interface/arrow_array.hpp"
+
+namespace sparrow
+{
+    /**
+     * Convert an array_data vector to an ArrowArray vector.
+     *
+     * @param ads The array_data vector to convert.
+     * @return The converted children.
+     */
+    std::vector<arrow_array_shared_ptr> children_from_array_data_vec(std::vector<array_data>&& ads);
```

Comment:
I'm  not a fan of the naming of these functions, it's kind of redundant.
Taking this specific function as an example, maybe `to_arrow_array_vec(whatever_range_of_array_data)` would be better? I mean at reading site the "from" side of these names isnt really helpful.
Also it's best if the names reflect the general idea instead of the specifics, so that variations in arguments can have the same name.

Also, why is "children" mentioned here? The function is said to just be a conversion isnt it?